microservice/src/test/java/de/gsi/microservice/concepts/RestBehindRouterEvaluation.java

```diff
@@ -0,0 +1,211 @@
+            //            handle.send(socket, ZFrame.MORE | ZFrame.REUSE);
+            //            socket.send((byte[]) null, 0);
+            return null;
+        }
+
+        return handle;
+    }
+
+    private static ZFrame getRequest(final Socket socket) {
+        // TODO: add further safe-guards if called for a socket with no data pending
+
+        // Get [id, ] message on client connection.
+        final ZFrame handle = ZFrame.recvFrame(socket);
+        if (handle == null || bytesToLong(handle.getData()) == 0) {
+            return null;
+        }
+
+        System.err.println("received Request ID = " + handle.toString() + " - more = " + handle.hasMore()); //  Professional Logging(TM)
+        if (!handle.hasMore()) {
+            //  Close erroneous connection to browser
+            handle.send(socket, ZFrame.MORE | ZFrame.REUSE);
+            socket.send((byte[]) null, 0);
+            return null;
+        }
+
+        // receive request
+        return ZFrame.recvFrame(socket);
+    }
+
+    private static void handleRouterSocket(final Socket router, final Socket stream) {
+        System.err.println("### called handleRouterSocket");
+        // Get [id, ] message on client connection.
+        ZFrame handle;
+        if ((handle = getConnectionID(router)) == null) {
+            // did not receive proper [ID, null msg] frames
+            return;
+        }
+
+        final ZFrame request = getRequest(router);
+        if (request == null) {
+            return;
+        }
+        if (Arrays.equals(ZERO_MQ_HEADER, request.getData())) {
+            router.sendMore(handle.getData());
+            router.send(ZERO_MQ_HEADER);
+            System.err.println("received ZeroMQ message more = " + request.hasMore());
+
+        } else {
+            System.err.println("received other (HTTP) message");
+            System.err.println("received request = " + request + " more? " + request.hasMore());
+            return;
+        }
+
+        //        handleStreamHttpSocket(router, handle);
+        // received router request
+        while (request.hasMore()) {
+            // receive message
+            final byte[] message = router.recv(0);
+            final boolean more = router.hasReceiveMore();
+            System.err.println("router msg (" + (more ? "more" : "all ") + "): " + Arrays.toString(message) + "\n      - string: '" + new String(message) + "'");
+
+            //handleStreamHttpSocket(router);
+            // Broker it -- throws an exception (too naive implementation?)
+            //stream.send(message, more ? ZMQ.SNDMORE : 0);
+            if (!more) {
+                break;
+            }
+        }
+    }
+
+    private static void handleStreamHttpSocket(Socket httpSocket, ZFrame handlerExt) {
+        // Get [id, ] message on client connection.
+        ZFrame handler = handlerExt;
+        if (handler == null) {
+            if ((handler = getConnectionID(httpSocket)) == null) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)